templates/deployments.yaml

```diff
@@ -982,13 +982,18 @@ spec:
     {{- end }}
 
 {{- range .Values.st2.packs.sensors }}
+  {{- $sensor := omit $.Values.st2sensorcontainer "name" "ref" "postStartScript" }}
+  {{- range $key, $val := . }}
+    {{- $_ := set $sensor $key $val }}
+  {{- end }}
```

Comment:
This applies the default. I looked at using one of the `merge` functions to generate the `$sensor` dictionary, but all of the `merge` functions are deep merge functions which is undesirable. Using `set` like this allows us to replace keys instead of deep merging them.